### ISSUE TITLE
Coroutines in C with Arbitrary Arguments

### ISSUE THREAD

**Issue by aliveyun** [zelang-dev/c-coroutine#3]
Coroutine with arbitrary parameters in C
Not usually only a single predefined type of function is allowed to be executed as a coroutine. For example, both libtask and libcoro require coroutines to have the following prototype:

void foo(void *arg);

**Comment by TheTechsTech** [comment]
What? You for real? 
> `void *arg`

That is the very means of passing arbitrary parameters. You know this is **C**, a low level language. 

I am going to take that you very new to programming, or very young. 

And don't know the origins of GoLang. Nor the difference between source code and the binary output 
and the layers of instructions/steps taking place between them.

You might be better off putting questions/issues on https://stackoverflow.com/, and then read the release notes of 
**GoLang 1.5** https://go.dev/doc/go1.5#implementation, where they stop building there language with **C**!!!

You need to post at actual issue you having, not your misplaced/lack of understanding of an language. 
There are many Computer Science courses on **YouTube**.

**Comment by TheTechsTech** [comment]
You using the **C** language, you are responsible for casting or creating another function to do whatever. You need to wrap it with another. 

You should not be using **C**, use a different language instead. Since you not aware of that. 

The function signature is that way intentionaly. You can pass whatever means of parameters you want. 

You just don't know how to, which means you shouldn't be using this package or the **C** language in general, it's not about teaching you.

**Comment by aliveyun** [comment]
You should refer to libdill, which can implement multi-parameter functions, but libdill does not support windows. C++ can also be implemented.
#define go_mem(fn) \
    std::thread([=] { \
        fn; \
    }).detach()

#define go(fn) go_mem(fn)

**Comment by TheTechsTech** [comment]
>  You should refer to libdill, which can implement multi-parameter functions, but libdill does not support windows. C++ can also be implemented.
#define go_mem(fn)
std::thread([=] {
fn;
}).detach()
#define go(fn) go_mem(fn)

You funny, the reason for this library, as I said, you need to learn **"how to right code"**, this is **C**. Stop just trying to copy!!! Learn what's happening under the hood.

Read the readme here again, look at the examples.
